src/uct/api/v2/uct_v2.h

```diff
@@ -963,6 +976,31 @@ ucs_status_t uct_ep_connect_to_ep_v2(uct_ep_h ep,
                                      const uct_ep_addr_t *ep_addr,
                                      const uct_ep_connect_to_ep_params_t *params);
 
+
+/**
+ * @ingroup UCT_MD
+ *
+ * @brief Compare two remote keys
+ *
+ * The comparison result is < 0 if rkey1 is lower than rkey2, 0 if they
+ * are equal or > 0 if rkey1 is greater than rkey2. The remote key greater-than
+ * semantic is arbitrary and implementation dependent, nevertheless it's
+ * guaranteed to be consistent (transitive and antisymmetric). It can be used,
+ * for example, to perform an efficient binary search in a sorted array of
+ * remote keys.
+ *
+ * @param[in]  component  Component to use for the comparison
+ * @param[in]  rkey1      First rkey to compare
+ * @param[in]  rkey2      Second rkey to compare
+ * @param[in]  params     Additional parameters for comparison
+ * @param[out] result     Result of the comparison
+ *
+ * @return Error code or UCS_OK if result contains the comparison result
```

Comment:
What UCP will do if error returned ? What does unsupported means ?

Reply:
The proposal is to have UCP immediately return that error.
The unsupported below is to be replaced by actual implementation after this PR is merged.

Reply:
what UCP will return if underlaying layer reports unsuported

Reply:
please check UCP side response: https://github.com/openucx/ucx/pull/9289#discussion_r1306987794